deployment/Triton/models/monai_covid/1/model.py

```diff
@@ -0,0 +1,166 @@
+import torch
+import torch.backends.cudnn as cudnn
+
+from monai.inferers.inferer import SimpleInferer
+from monai.transforms import Compose
+from monai.transforms import (Activations,
+                              AddChannel,
+                              AsDiscrete,
+                              CropForeground,
+                              LoadImage,
+                              Lambda,
+                              ScaleIntensityRange,
+                              ToNumpy,
+                              ToTensor,
+                              Resize)
+
+import triton_python_backend_utils as pb_utils
+
+
+logger = logging.getLogger(__name__)
+
+
+class TritonPythonModel:
+    """
+    Your Python model must use the same class name. Every Python model
+    that is created must have "TritonPythonModel" as the class name.
+    """
+
+    def initialize(self, args):
+        """
+        `initialize` is called only once when the model is being loaded.
+        Implementing `initialize` function is optional. This function allows
+        the model to intialize any state associated with this model.
+        """
+
+        # load model configuration
+        self.model_config = json.loads(args['model_config'])
+
+        # create inferer engine and load PyTorch model
+        inference_device_kind = args.get('model_instance_kind', None)
+        logger.info(f"Inference device: {inference_device_kind}")
+
+        self.inference_device = torch.device('cpu')
+        if inference_device_kind is None or inference_device_kind == 'CPU':
+            self.inference_device = torch.device('cpu')
+        elif inference_device_kind == 'GPU':
+            inference_device_id = args.get('model_instance_device_id', '0')
+            logger.info(f"Inference device id: {inference_device_id}")
+
+            if torch.cuda.is_available():
+                self.inference_device = torch.device(f'cuda:{inference_device_id}')
+                cudnn.enabled = True
+            else:
+                logger.error(f"No CUDA device detected. Using device: {inference_device_kind}")
+
+        # create pre-transforms
+        self.pre_transforms = Compose([
+            LoadImage(reader="NibabelReader", image_only=True, dtype=np.float32),
+            AddChannel(),
+            ScaleIntensityRange(a_min=-1000, a_max=500, b_min=0.0, b_max=1.0, clip=True),
+            CropForeground(margin=5),
+            Resize([192, 192, 64], mode="area"),
+            AddChannel(),
+            ToTensor(),
+            Lambda(func=lambda x: x.to(device=self.inference_device)),
+        ])
+
+        # create post-transforms
+        self.post_transforms = Compose([
+            Lambda(func=lambda x: x.to(device="cpu")),
+            Activations(sigmoid=True),
+            AsDiscrete(threshold_values=True, logit_thresh=0.5),
+            ToNumpy(),
+        ])
+
+        self.inferer = SimpleInferer()
+
+        self.model = torch.jit.load(
```

Comment:
with this approach, we are not really taking advantages of triton (GPU management)
It's almost same running everything native in pytorch..

otherwies the only benefit in this approach that i see here is... you can load many of such models in triton and trigger inference through http/grpc

Reply:
otherwise the example code looks ok to me..

Reply:
Good observation(s). The goal was to first show how the Monai (Pytorch) model could be implemented in Triton.  Your suggestion could be another project where we show how to take more advantage of Triton and some of those things are planned. Could you suggest others in the issues and then we can track adding those capabilities.